interview-questions.md

```diff
@@ -8,44 +8,44 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is a gem?
 
-Your answer:
+Your answer: a gem is a repository to save the data you worked on
 
-Researched answer:
+Researched answer: A gem is library or chunk of code for a certain functionality 
 
 2. What does the acronym CRUD stand for?
 
-Your answer:
+Your answer: create, read, utilize, delete
 
-Researched answer:
+Researched answer: it stands for create, update , and delete
 
 3. What is Ruby on Rails?
 
-Your answer:
+Your answer: ruby on rails is a framework adding more functionality for ruby for web development and database development
 
-Researched answer:
+Researched answer: Ruby on Rails is a backend web app framework made in Ruby
 
 4. What is a relational database? Are there other kinds of databases?
 
-Your answer:
+Your answer: a relational database is a database where the components communicate with each other , there are different types of databases
 
-Researched answer:
+Researched answer: Relational databases are a kinds of database that record and organize data  with defined relationships for easier acccess to them
 
 5. What are primary keys? Why are they important?
 
-Your answer:
+Your answer:primary keys are data pairs that help differentiate and invoke values from the data whether it be arrays, objects or hashes
 
-Researched answer:
+Researched answer: primary keys are unique fields in a table to identify each row
 
 ## Looking Ahead: Terms for Next Week
 
 Research and define the following terms to the best of your ability.
 
-1. RESTful routes:
+1. RESTful routes: RESTful routing is a set of protocols used in multiple different languages to create working, reusable routes
 
-2. JSON:
+2. JSON: JavaScript Object Notation (JSON) is a standard text-based format for representing structured data based on JavaScript object syntax. It
 
-3. ERB:
+3. ERB: returns a string in the same character encoding as the input string.
 
-4. Params:
+4. Params:  parameter is a  variable in computer programming l that is used to pass information between functions or methods
 
-5. API:
+5. API: software that lets applications communicate with each other
```

Comment:
I would like to see you expand on your answers here.  Remember that these are interview questions and the more you can explain your answers (use examples!), the more interested and knowledgable you will look.  This is the bare minimum for these questions.